types/fraction.go

```diff
@@ -0,0 +1,14 @@
+package types
+
+type Fraction struct {
+	Numerator   int
+	Denominator int
```

Comment:
Let's make this fixed width (int64) to avoid surprises of people run that on 32 bit platforms one day.

I also don't think we need signed integers, but then again Len() returns an `Int` 🤷‍♂️ And thoughts on signedness?

Reply:
I tend to think of things like `Fraction` as a generic util we might one day use for something else, so I wanted to go with some sort of more "generic" signed type, but I'm not insisting on that.

Reply:
Alright, good. We can create a UFraction later on oif we want.

In this case we should have a `isNegative()` method that allows callers to easily convert such cases to errors.

Reply:
So now we would be adding a call to `isNegative` to every use we have? When you put it like that, we're just creating a bunch more boilerplate...

Reply:
When a negativ cost is configured, very strange strings will happen (gas will be recharged). I think this should be prevented. Now you can blame the caller for that. But we can also support the caller when doing something stupied.

I'd do one of the two:
1. Error at runtime for genative values
2. Change the type to only support non-negative values

Reply:
Yeah, I think I really prefer 2. Just pushed that.